File: dotnet/src/MyDotey.SCF.Simple/Type/String/StringToLongConverter.cs
using System;

namespace MyDotey.SCF.Type.String
{
    /**
     * @author koqizhao
     *
     * May 21, 2018
     */
    public class StringToLongConverter : StringConverter<long?>
    {
        public static readonly StringToLongConverter DEFAULT = new StringToLongConverter();

        public override long? Convert(string source)
        {
            if (string.IsNullOrWhiteSpace(source))
                return null;

            return long.Parse(source);
        }
    }
}

File: java/scf-simple/src/main/java/org/mydotey/scf/type/string/StringToLongConverter.java
package org.mydotey.scf.type.string;

/**
 * @author koqizhao
 *
 * May 21, 2018
 */
public class StringToLongConverter extends StringConverter<Long> {

    public static final StringToLongConverter DEFAULT = new StringToLongConverter();

    public StringToLongConverter() {
        super(Long.class);
    }

    @Override
    public Long convert(String source) {
        return Long.parseLong(source);
    }

}
